### ISSUE TITLE
Fatal error during greenlock init

### ISSUE THREAD

**Issue by egallup02360** [Ylianst/MeshCentral#812]
Running into issues trying to use let's encrypt for certificates.

Console Output During Run:
```
FAKE CERTIFICATES (for testing) only

Fatal error during greenlock init:
Cannot read property 'runAsProduction' of undefined

    at ChildProcess.exithandler (child_process.js:295:12)
    at ChildProcess.emit (events.js:228:7)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5) {
  killed: false,
  code: 1,
  signal: null,
  cmd: '/usr/bin/node /root/meshcentral/node_modules/meshcentral/meshcentral --launch 19096'
}
ERROR: MeshCentral failed with critical error, check MeshErrors.txt. Restarting in 5 seconds...
```

config.json
```
{
 "settings": {
     "cert": "mydomain",
     "redirPort": 80
 },
 "letsencrypt": {
     "email": "email@email.com",
     "names": "mydomain",
     "rsaKeySize": 3072,
     "production": false
 }
}
```

Edit: Node.js version 12.14.1

**Comment by OutbackMatt** [comment]
I think this is related to the greenlock version installed

I was researching this very recently, after having the same issue
https://github.com/Ylianst/MeshCentral/issues/629

**Comment by asasin114** [comment]
What version of Node and what version of Mesh Central are you running? I had a lot of cert issues back when things switched over to the new version of Greenlock.

**Comment by Ylianst** [comment]
I am traveling next week, so, not going to be able to fix things as much. If you installed MeshCentral many months back and recently decided to configure Let's Encrypt, the problem may be that an older GreenLock is present (or something).

The fest thing to do is to go in the folder above "node_modules" and do (assuming Linux):

```
mv node_modules node_modules_back
npm install meshcentral
```

And try again... You get fresh installation with all latest libraries. Note that all the important data is in meshcentral-data, meshcentral-files, so you can just reset the node_modules and it's ok.

If that does not solve anything, I just looked at the code and I can see where this error should happen (in the letsencrypt.js module in MeshCentral). I would have to work on debugging this. First, let me know if a node_modules reset solved it. -Thanks.

**Comment by OutbackMatt** [comment]
This worked for me - thanks

**Comment by egallup02360** [comment]
I'm running version 0.4.7-i of meshcentral and Node 12.14.1
This was all a fresh install yesterday when I ran into this issue. I just tried what you suggested and no dice.

I'm running this on digitalocean so I just rebuilt the server from a clean ubuntu 18.04 image (I was running on a prebuilt image of ubuntu with node and npm preinstalled yesterday). After the rebuild, I installed node 10.18.1 then meshcentral, ran once to generate data directories, then turned on let's encrypt in the config.json. Same error as before.

**Comment by egallup02360** [comment]
So the issue certainly appears to be in greenlock (probably in how meshcentral is interacting with it). I downgraded greenlock with `npm install greenlock@^v3` and it's working perfectly now.

Figured this would be helpful information while debugging.

**Comment by Ylianst** [comment]
Oohhh!!! I just noticed that [GreenLock 4.0](https://www.npmjs.com/package/greenlock) was released and this likely breaks MeshCentral. I will published a new version that explicitly installs GreenLock  3.x until I get this fixed. I will do this today.

**Comment by Ylianst** [comment]
Just published MeshCentral v0.4.7-j where I explicitly ask for GreenLock 3.1.5. This should fix this until I get around to supporting GreenLock 4.

**Comment by egallup02360** [comment]
Downgrading to greenlock 3.1.5 has fixed this.
